### DISCUSSION TITLE
Python-SDK: Streaming from browser mic?

### DISCUSSION THREAD

**Discussion by gziz** [deepgram/community#213] [General help] [answered]
Is it possible to use the python-sdk for streaming audio, where the audio comes from the browser microphone?
Ultimately I would like my web application to have a python backend. Hope you can share an example, in case you have one.
Thanks!

**Comment by lukeocodes** [comment, marked as answer]
Hi @gziz there are a few of ways to achieve this. 

You could connect to Deepgram's websocket in the browser (using JS), and send your audio, and use your python application as a callback endpoint to do what you want to do for a user.

You could also connect to your own python websocket, and then send that on to the deepgram websocket without a callback. Response messages recieved to do what you want to do for a user.

**Reply by gziz** [reply]
Got it @lukeocodes. 
Thanks for the reply. 😃
